src/storage/hummock_sdk/src/sstable_info_ref.rs

```diff
@@ -0,0 +1,224 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::sync::{Arc, Weak};
+
+use risingwave_pb::hummock::{PbBloomFilterType, PbHummockVersion, PbSstableInfo};
+use tokio::task_local;
+
+use crate::change_log::EpochNewChangeLogCommon;
+use crate::key_range::KeyRange;
+use crate::sstable_info::SstableInfo;
+use crate::version::{HummockVersion, HummockVersionCommon, HummockVersionDeltaCommon};
+use crate::HummockSstableId;
+
+pub type SstableInfoType = SstableInfoRef;
+pub type EpochNewChangeLogType = EpochNewChangeLogCommon<SstableInfoType>;
+pub type HummockVersionType = HummockVersionCommon<SstableInfoType>;
+pub type HummockVersionDeltaType = HummockVersionDeltaCommon<SstableInfoType>;
+
+pub trait SstableInfoReader {
```

Comment:
Instead of implementing this trait, it's easier to use `impl Deref<Target = SstableInfo>`